Include/Rocket/Core/StringBase.inl

```diff
@@ -87,7 +87,7 @@ StringBase< T >::StringBase(size_type ROCKET_UNUSED_PARAMETER(max_length), const
 template< typename T >
 StringBase< T >::~StringBase()
 {
-	if (value != (T*)local_buffer)
+	if (!IsUsingLocalBuffer())
```

Comment:
Again, where is this "complexity" you are talking about? 

The interest is in knowing if the local buffer is in use or not, not how we know. Nothing is simpler than seeing a function name explicitly indicate that. It's not complex. "Complex" is having to decipher a cast and pointer comparison each time that initially looks like a deeper value comparison and having to work it out "oh that's actually what he's testing for" each time.  That *should* be obscured to use your terminology because it's an implementation detail and of no cost. Another implementation might indicate to itself that it's using the local buffer by setting the top bit of some other field like length or something wild or put the value pointer and buffer in a union to safe space. It's all an implementation detail.

So where is this complexity you talk about, I see a no cost abstraction that's more readable? If you just think the change isn't worth your time to apply because it's not of high value, I could support that, but I don't support not doing it because the code is now somehow more "complex'. It isn't to me. Overall, I don't see that you have demonstrated any complexity claim beyond saying "because I think so" but if you're still not convinced by my more detailed reasoning, by all means keep what you have. It's a small change and you are supporting it more than I will.